app/src/main/java/ch/epfl/sdp/blindly/settings/Settings.kt

```diff
@@ -106,20 +106,30 @@ class Settings : AppCompatActivity() {
         }
     }
 
-    //TODO in another branch
-    /*
-    override fun onBackPressed() {
-        if(radiusSlider.value != currentRadius.toFloat()) {
+    @RequiresApi(Build.VERSION_CODES.N)
+    override fun onResume() {
+        super.onResume()
+        viewModel.userUpdate()
+    }
 
+    @RequiresApi(Build.VERSION_CODES.N)
+    override fun onBackPressed() {
+        if (radiusSlider.value != currentRadius.toFloat()) {
+            viewModel.updateField(RADIUS, radiusSlider.value.toInt())
         }
-        if(ageRangeSlider.values != listOf(currentAgeRange[0].toFloat(), currentAgeRange[1].toFloat())) {
-
+        if (ageRangeSlider.values != listOf(
+                currentAgeRange[0].toFloat(),
+                currentAgeRange[1].toFloat()
+            )
+        ) {
+            viewModel.updateField(
+                AGE_RANGE,
+                listOf(ageRangeSlider.values[0].toInt(), ageRangeSlider.values[1].toInt())
```

Comment:
All this code remains untested. Can you think of a way to test it by clicking on the back button, or is it too complicated ?